### ISSUE TITLE
Osram Smart Plug Not changing state by button

### ISSUE THREAD

**Issue by F4bsi** [mozilla-iot/zigbee-adapter#118]
Hey,

if I use the button on the Osram Smart+ Plug to turn it off or on the state does not change in the gateway. Toggling it from the gateway still works, but it's in a reversed state (for example it shows "off" but it's on and vise versa).

Maybe there is a way to query the state? I have no idea how to debug this, so maybe you can help.

**Comment by dhylands** [comment]
It seems to depend on the device. Not all devices report their changes to the gateway or need to be configured to report their changes.

The Aeotec Smart Switch (which are ZWave - not zigbee but the point is the same) also doesn't report its changes by default, but has a config option to do do. This is documented in the Aeotec documentation for their device so I was able to add code to the classifier to change the configuration setting.

If the OSRAM device has a similar configuration setting, then we can set it. But very few manufacturers seem to document their configuration settings.

I'll check and see if I have one and see if it behaves the same.

**Comment by F4bsi** [comment]
Thanks for the clarification.

I just added the plug to an Philips hue gateway to see if it shows the changes when pressing the button and it works, so it does seem to work. Maybe it really is just a configuration thing?

**Comment by dhylands** [comment]
It could also bee that the zigbee adapter needs to listen for something different. I have an OSRAM smartplug, so I'll see if I can do some tests.

**Comment by riccardocossu** [comment]
I am facing the same issue; could a manual (through a button or function in the detail page) or periodic polling of the value from the device solve the problem? It would also probably be a simple change in just the UI (the APIs are already all there - one GET to get the value and then just update the internal state).

**Comment by mrstegeman** [comment]
No, this should not be something in the UI. A user should not be forced to manually poll property values.

**Comment by riccardocossu** [comment]
not property values, just a refresh of the component state; in the case of this plug it is just one property, but you get the idea.
But if it's not a UI change, would it be possibile for the addon to do it in the background?

**Comment by mrstegeman** [comment]
The component state comes directly from the property value, which comes from the add-on, not the gateway. So yes, that's exactly what I'm saying. The add-on needs to keep the property state up to date in the background.

This adapter _does_ do that, but appears to fail for certain devices.

**Comment by riccardocossu** [comment]
Ok is there any way I can provide more feedback about what's happening with my plug (debug log, API calls...)?
